typing/ctype.ml

```diff
@@ -2638,15 +2609,12 @@ let unify3_var env t1' t2 t2' =
 
 let rec unify (env:Env.t ref) t1 t2 =
   (* First step: special cases (optimizations) *)
-  if t1 == t2 then () else
```

Comment:
It looks like this fast path has been removed. Could this affect performance? In the new code it should still be possible to return early if `t1 == t2` and/or if `get_desc t1 == get_desc t2`.

Reply:
Actually there is a call to `eq_type t1 t2` at the beginning of `unify_eq` called below, so maybe it's fine. I note that `eq_type` checks `get_desct t1 == get_desc t2`, but not `t1 == t2` first, I wonder if this makes a difference.

Reply:
The point is that, since normalization is done on access rather than on the node itself, the probability that two type nodes are physically equal gets lower (i.e., now normalization changes the `Tlink` inside the node, but does not change the node we are looking at, as it was the case with `repr`).
We could still add a fast path to `Types.eq_type`. This is an extra test, but this may avoid two calls to `repr` when successful.